Couchers-org/couchers
# Web push notifications

This PR listens for the push event from the backend and displays the notifications for different actions.

Right now this will only work in Chrome, Firefox or other browsers that support the web push API (not safari, but mostly everything else).

![Screenshot 2024-09-10 at 9 54 15 PM](https://github.com/user-attachments/assets/3c70228b-aad4-426d-8678-6011ef9137bb)

![Screenshot 2024-09-10 at 9 54 23 PM](https://github.com/user-attachments/assets/7267e105-5d26-4edc-919d-8737fb633c40)

<img width="425" alt="Screenshot 2024-09-11 at 9 16 42 PM" src="https://github.com/user-attachments/assets/7e6fcd69-ba29-4e54-89a7-69634f49d16b">

**HOW TO TEST:**

- You need to run both front and backend locally (so push notifications are sent from the same environment they are received)
- Generate protos locally if needed
- Don't use Safari, Chrome works well
- Go to Feature Preview and allow notifications there.
- I decided to test messaging push notifications, you can do something else. To test messaging notifications you need two couchers accounts. Log in to one in the regular browser, the other in an incognito window.
- Message yourself from the incognito window to the other (push notifications don't work in incognito mode)
- Check non-incognito window, push notification should show.
- Reloading the page also triggers a push notification, if you are lazy.

**Web frontend checklist**
- [ x ] Formatted my code with `yarn format`
- [ x ] There are no console warnings when running the app
- [ N/A ] Added any new components to storybook
- [ x ] Added tests where relevant
- [ x ] All tests pass
- [ x ] Clicked around my changes running locally and it works
- [ x ] Checked Desktop, Mobile and Tablet screen sizes
